server/src/tables.rs

```diff
@@ -40,240 +47,361 @@ use datafusion::physical_plan::{
+    /// retrieved from [`catalog::METADATA_SQLITE_NAME`].
+    fn compute_keys_using_fields_and_tags(
+        &self,
+        table_name: &str,
+        columns: &Option<Vec<usize>>,
+        tag_predicates: &[(&str, &str)],
+    ) -> Result<Vec<TimeSeriesId>> {
+        // Compute the location of the database containing the metadata.
+        let database_path = {
+            // unwrap() is safe as read() only fails if the RwLock is poisoned.
+            let catalog = self.context.catalog.read().unwrap();
+            catalog.data_folder_path.join(catalog::METADATA_SQLITE_NAME)
+        };
+
+        // Construct the two queries that extract the field columns in the table
+        // being queried and the hashes of the multivariate time series in the
+        // table with tag values that match the tag values in the user's query.
+        let query_field_columns = if columns.is_none() {
+            format!(
+                "SELECT column_index FROM model_table_field_columns WHERE table_name = '{}'",
+                table_name
+            )
+        } else {
+            let column: Vec<String> = columns
+                .clone()
+                .unwrap()
+                .iter()
+                .map(|column| format!("column_index = {}", column))
+                .collect();
+
+            format!(
+                "SELECT column_index FROM model_table_field_columns WHERE table_name = '{}' AND {}",
+                table_name,
+                column.join(" OR ")
+            )
+        };
+
+        let query_hashes = {
+            if tag_predicates.is_empty() {
+                format!("SELECT hash FROM {}_tags", table_name)
+            } else {
+                let predicates: Vec<String> = tag_predicates
+                    .iter()
+                    .map(|(tag, tag_value)| format!("{} = {}", tag, tag_value))
+                    .collect();
+
+                format!(
+                    "SELECT hash FROM {}_tags WHERE {}",
+                    table_name,
+                    predicates.join(" AND ")
+                )
+            }
+        };
+
+        // Retrieve the hashes using the queries and reconstruct the keys.
+        self.compute_keys_using_sqlite_database(&database_path, &query_field_columns, &query_hashes)
+            .map_err(|error| DataFusionError::Plan(error.to_string()))
     }
 
-    fn binary_expr(&self, left: Expr, op: Operator, right: Expr) -> Expr {
-        Expr::BinaryExpr {
-            left: Box::new(left),
-            op,
-            right: Box::new(right),
+    // TODO: Move to the metadata component when it exists.
+    /// Compute the 64-bit keys of the univariate time series to retrieve from
+    /// the storage engine using the two queries constructed from the fields,
+    /// tag, and tag values in the user's query. Returns a [`RusqliteResult`] if
+    /// the data cannot be retrieved from [`catalog::METADATA_SQLITE_NAME`].
+    fn compute_keys_using_sqlite_database(
```

Comment:
This is more a general question than a thing to fix right now. Should we avoid writing `sqlite` in the code and instead just refer to it as the metadata database? One problem with using `sqlite` is that it makes it harder to change databases in the future, if needed.

Reply:
I agree, and as stated in [comment](https://github.com/ModelarData/ModelarDB-RS/pull/37#discussion_r986723214) I have changed the use of `catalog::METADATA_SQLITE_NAME` in doc comments to metadata database. Due to this comment I have also changed the name of `compute_keys_using_sqlite_database()` to `compute_keys_using_metadata_database()`. If we want to support additional RDBMSs in the future it may be worth looking into replacing [`rusqlite`](https://crates.io/crates/rusqlite/) with a RDBMS agnostic crate like [`SQLx`](https://crates.io/crates/sqlx) or [`Diesel`](https://crates.io/crates/diesel), although a full ORM like Diesel is probably overkill for our use case.